frontend/components/book-table-page/BooktablePage.tsx

```diff
@@ -0,0 +1,214 @@
+import { ControlledInput } from '@/components/common/ControlledInput';
+import { ControlledTestArea } from '@/components/common/ControlledTestArea';
+import { ControlledSelect } from '@/components/common/ControlledSelect';
+import { Button } from '@chakra-ui/react';
+import { useForm } from 'react-hook-form';
+import { InputsContainer } from '@/components/Take-away-page/component/InputsContainer';
+import { useEffect, useState } from 'react';
+import { zodResolver } from '@hookform/resolvers/zod';
+import * as zod from 'zod';
+import VerifyOtpModal from '@/components/book-table-page/VerifyOtpModal';
+import useTimer from './useTimer'; 
+import { Restaurant } from '@/types';
+import { isTimeWithinRange } from './timeUtils';
+
+interface BooktablePageProps {
+  restaurant: Restaurant;
+}
+
+type FormData = {
+  name: string;
+  phone: string;
+  time: string;
+  guests: string;
+  preference: string;
+  notes: string;
+  email: string;
+  date: string;
+};
+
+export function BooktablePage({ restaurant }: BooktablePageProps) {
+
+  const requiredField = zod.string().min(1, { message: 'Required Field' });
+const phoneSchema = zod.string().regex(/^\d{9}$/, { message: 'Phone number invalid' });
+
+const FormDataSchema = zod.object({
+  name: requiredField,
+  phone: phoneSchema,
+  date: requiredField,
+  time: requiredField,
+  guests: requiredField,
+  email: requiredField.email(),
+}).superRefine((data, context) => {
+  const selectedDate = new Date(data.date); 
+  const dayOfWeek = selectedDate.getDay();
+  const isWeekday = dayOfWeek >= 1 && dayOfWeek <= 4; 
+  const weekdayTimeRange = restaurant.Weekdaytime;
+  const weekendTimeRange = restaurant.Weekandtime;
+
+  const isTimeValid = isWeekday
+    ? isTimeWithinRange(data.time, weekdayTimeRange.start, weekdayTimeRange.end)
+    : (
+        isTimeWithinRange(data.time, weekdayTimeRange.start, weekdayTimeRange.end) ||
+        isTimeWithinRange(data.time, weekendTimeRange.start, weekendTimeRange.end)
+      );
+
+  if (!isTimeValid) {
+    context.addIssue({
+      code: zod.ZodIssueCode.custom,
+      message: 'Time is outside of restaurant operating hours',
+      path: ['time'], 
+    });
+  }
+
+  if (restaurant.blacklist.includes(data.phone)) {
+    context.addIssue({
+      code: zod.ZodIssueCode.custom,
+      message: 'Your phone number is in the blacklist',
```

Comment:
Make this to say internet connection failed. We don't want them to know we have a blacklist and try to figure out how to get around